design/0001-frr.md

```diff
@@ -310,3 +310,58 @@ releases is proposed:
    MetalLB is deprecated from this point onwards. FRR is no longer
    limitied by feature parity with the current native implementation of BGP and
    additional functionality may be developed.
+
+## Test Coverage
+
+The current e2e tests setup a bgp peer to all the nodes inside an FRR container, and try to hit the service.
+
+In order to guarantee parity between the two implementations, the coverage of the e2e tests must be
+extended.
+
+### Test strategy
+
+Every time we test an exposed service, we must verify that:
+
+* the service is reacheable from outside
+* the nodes advertising the service ip are the expected ones
+* all the nodes are paired with the expected bgp peers
```

Comment:
I guess there are some things we can read from the peer to ensure the connection is as expected. One thing for example is the connection state but maybe there are others? We currently don't do that but it could be useful.

Reply:
Yes, the connection state is already part of my PR https://github.com/metallb/metallb/pull/944 , plus there I check that routes are set as we expect both in FRR and inside the container.

Reply:
Yes, I started reviewing your PR after this. Good idea!